### PR TITLE
Add resource x509_request

### PR DESCRIPTION
**Pull request by juju482**
### Description

Hello, this is another contribution from the french organization [Institut National de l'Audiovisuel](https://institut.ina.fr)

It add a new resource with documentation & tests :

    openssl_x509_request : Create x509 Certificate Request

I also change the kitchen files because the cookbook isn't working with chef14 (dhparam is now directly included in the chef client).

### Issues Resolved

https://github.com/chef-cookbooks/openssl/issues/60

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 driver:
   name: dokken
   privileged: true # because Docker and SystemD/Upstart
-  chef_version: <%= ENV['CHEF_VERSION'] || 'current' %>
```
Comment (tas50) [inline kitchen.dokken.yml]: The goal here was to default to testing the current release, but fallback if we wanted to. If you wan to test Chef 13 you should be specifying the version

**Comment by tas50** [comment]
Thanks for putting this together. It was actually on our todo list to get a resource like this in core Chef.

File: kitchen.dokken.yml (hunk 1 above)
Comment (juju482) [reply kitchen.dokken.yml]: Hello Tim. I didn't realise the main goal of this before i edited it... And I should ! Nice feature. 

I'll correct my work very soon. Thanks for the quick review.

**Comment by tas50** [comment]
That sounds like a good feature to have as well.

**Pull request merged**